pyam/utils.py

```diff
@@ -585,3 +585,11 @@ def reduce_hierarchy(x, depth):
     _x = x.split('|')
     depth = len(_x) + depth - 1 if depth < 0 else depth
     return '|'.join(_x[0:(depth + 1)])
+
+
+def get_variable_components(x, depths):
+    """Return (or merge and return) the variable component(s) in regard to
```

Comment:
Please use a one-line summary, then an empty line, then the rest of the description, see the [numpydoc docs](https://numpydoc.readthedocs.io/en/latest/format.html)